Create rule S7470 (#4875)
SonarSource/rspec: rules/S7470/python/rule.adoc

This rule raises an issue when `RDD.groupByKey` is used in conjuction with `RDD.mapValues`
and a commutative and associative function instead of `RDD.reduceByKey`.

== Why is this an issue?

The PySpark API offers multiple ways of performing aggregation. 

> When performing aggregations, data is usually shuffled between partitions of course, 
> this shuffling and its associated cost are needed to compute the result correctly.

Reviewer: ```suggestion
When performing aggregations, data is usually shuffled between partitions.
This shuffling is needed to compute the result correctly. It has an associated cost that can impact performance.
```
